### PR TITLE
Display failed hook in React

### PR DESCRIPTION
**Pull request by bastienmalatier**
#411  Summary

If we have a failed hook in react, we don't see it in the current html report. With this modifications, we want them to appear in it.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] The change has been ported to Java.
- [ ] The change has been ported to Ruby.
- [x] The change has been ported to JavaScript.
- [ ] The change has been ported to Go.
- [ ] The change has been ported to .NET.
- [x] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the CHANGELOG accordingly.

### CODE REVIEW

File: query/javascript/src/Query.ts [hunk 1]
```diff
@@ -157,4 +167,51 @@ export default class Query {
     | undefined {
     return this.stepMatchArgumentsListsByPickleStepId.get(pickleStepId)
   }
+
+  public getBeforeHookSteps(
+    pickleId: string
+  ): ReadonlyArray<messages.TestCase.ITestStep> {
+    const testCase = this.testCaseByPickleId.get(pickleId)
+
+    // const hookSteps = testCase.testSteps.filter((step) => step.hookId)
```
Comment (aslakhellesoy) [inline query/javascript/src/Query.ts:176]: Remove commented-out code

File: query/javascript/src/Query.ts [hunk 2]
```diff
@@ -157,4 +167,51 @@ export default class Query {
...
+    const hookSteps: messages.TestCase.ITestStep[] = []
+    let pickleStepFound = false
```
Comment (aslakhellesoy) [inline query/javascript/src/Query.ts:178]: Why is this declared outside the loop?

File: query/javascript/src/Query.ts [hunk 3]
```diff
@@ -157,4 +167,51 @@ export default class Query {
...
+
+    for (const step of testCase.testSteps) {
```
Comment (aslakhellesoy) [inline query/javascript/src/Query.ts:180]: Can you use `Array#filter` to build `hookSteps` instead?

File: query/javascript/src/Query.ts [hunk 4]
```diff
@@ -157,4 +167,51 @@ export default class Query {
...
+      }
+      if (step.hookId && !pickleStepFound) {
+        hookSteps.push(step)
+      }
+    }
+
+    return hookSteps
+  }
+
+  public getAfterHookSteps(
```
Comment (aslakhellesoy) [inline query/javascript/src/Query.ts:192]: Can you extract the duplicated code between `getAfterHookSteps` and `getBeforeHookSteps` to a separate function?

File: query/javascript/src/Query.ts [hunk 5]
```diff
@@ -157,4 +167,51 @@ export default class Query {
...
+      if (step.hookId && pickleStepFound) {
+        hookSteps.push(step)
+      }
+    }
+
+    return hookSteps
+  }
+
+  public getTestStepResults(testStepId: string): messages.ITestStepResult[] {
+    const testStepResults = this.testStepResultsbyTestStepId.get(testStepId)
```
Comment (aslakhellesoy) [inline query/javascript/src/Query.ts:213]: Intermediate variable not needed - just return the result

**Comment by vincent-psarga** [comment]
@aslakhellesoy we were discussing with Bastien and one question came: should we always display the Hook Step if there's an attachment or only when it failed ?
My guess is that we should always display attachments.

**Pull request merged**